client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/IndexLifecycleExplainResponse.java

```diff
@@ -73,10 +73,10 @@
         PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), MANAGED_BY_ILM_FIELD);
         PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), POLICY_NAME_FIELD);
         PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), LIFECYCLE_DATE_MILLIS_FIELD);
-        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), PHASE_FIELD);
-        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), ACTION_FIELD);
-        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), STEP_FIELD);
-        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), FAILED_STEP_FIELD);
+        PARSER.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), PHASE_FIELD);
+        PARSER.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), ACTION_FIELD);
+        PARSER.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), STEP_FIELD);
+        PARSER.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), FAILED_STEP_FIELD);
```

Comment:
We might want to have a smarter check here I think. We only expect the case where all the fields are `null`, we never expect only one of the fields to be null so I think we should add a check int he constructor that ensure if one of the phase, action, step, failed_step fields are null they all are and otherwise the all have a value throwing an exception if the check fails.